llm_toolkit/models.py

```diff
@@ -263,6 +263,78 @@ class GPT4o(GPT):
   name = 'gpt-4o'
 
 
+class AzureGPT(GPT):
+  """Azure's GPT model."""
+
+  name = 'gpt-3.5-turbo-azure'
+
+  # ================================ Prompt ================================ #
+  def estimate_token_num(self, text) -> int:
+    """Estimates the number of tokens in |text|."""
+    # https://cookbook.openai.com/examples/how_to_count_tokens_with_tiktoken
+    try:
+
+      encoder = tiktoken.encoding_for_model(self.name.replace('-azure', ''))
+
+    except KeyError:
+      logger.info('Could not get a tiktoken encoding for %s.', self.name)
+      encoder = tiktoken.get_encoding('cl100k_base')
+
+    num_tokens = 0
+    for message in text:
+      num_tokens += 3
+      for key, value in message.items():
+        num_tokens += len(encoder.encode(value))
+        if key == 'name':
+          num_tokens += 1
+    num_tokens += 3
+    return num_tokens
```

Comment:
This seems similar to the GPT classes (it's a wrapper of OpenAI's GPT). Can you re-use the existing classes instead?

The only change seems to be getting the client.

Reply:
Yes! It would be better. I have updated it.:)